mitmproxy/http.py

```diff
@@ -795,6 +789,22 @@ def port(self) -> int:
     @port.setter
     def port(self, port: int) -> None:
         self.data.port = port
+        self._update_host_and_authority()
+
+    def _update_host_and_authority(self) -> None:
+        val = url.hostport(
+            self.scheme,
+            self.host,
+            # test_http.py::TestRequestCore::test_port sets port = b"foo" as the port
```

Comment:
Let's check that `port` is an `int` in `port.setter` and raise a ValueError if that's not the case. We can then adjust the test and skip this check here :)